backend/src/main/java/com/github/NicolaiKopka/web_page_services/user_pages/user_favorites/UserFavoritesService.java

```diff
@@ -30,23 +30,36 @@ public Collection<Movie> getAllFavoriteMoviesFromDbByUser(String username) {
 
         return userFavorites.getMovieIds().stream().map(movieDBApiConnect::getMovieById).toList();
     }
+
+    // TODO Returns immutable collections. Check why and fix
     public UserFavoritesSaveObject addMovieToFavorites(Integer movieId, String username) {
 
         MyUser user = myUserRepo.findByUsername(username).orElseThrow();
 
         try{
             UserFavoritesSaveObject userFavorites = userFavoritesRepo.findByUserId(user.getId()).orElseThrow();
-            // TODO any way to make return List mutable?
-            List<Integer> idList = new ArrayList<>(userFavorites.getMovieIds().stream().toList());
-            idList.add(movieId);
-            userFavorites.setMovieIds(idList);
+            List<Integer> movieIds = new ArrayList<>(List.copyOf(userFavorites.getMovieIds()));
+            if(movieIds.contains(movieId)) {
+                throw new IllegalArgumentException("Movie already in list");
+            }
+            movieIds.add(movieId);
+            userFavorites.setMovieIds(movieIds);
```

Comment:
I remember that we talked about that. What I ment was something like that.
Here:
```java
userFavorites.addMovieId(movieId);
```
And in the `UserFavoritesSaveObject `:
```
public UserFavoritesSaveObject {
    
    private List<Integer> movieIds = new ArrayList<>();

    ...

    public void addMovieId(Integer movieId) {
        if(movieIds.contains(movieId)) {
            throw new IllegalArgumentException("Movie already in list");
        }
        movieIds.add(movieId);
    }
}
```